sdks/java/core/src/main/java/org/apache/beam/sdk/coders/Coder.java

```diff
@@ -344,8 +344,8 @@ public NonDeterministicException(
 
     @Override
     public String getMessage() {
-      return String.format(
-          "%s is not deterministic because:%n  %s", coder, Joiner.on("%n  ").join(reasons));
+      String reasonsStr = Joiner.on(String.format("%n  ")).join(reasons);
```

Comment:
Should it be `Joiner.on("\n\t")`? Does the `%n` work at all in this context? Perhaps add a small test?